yb-voyager/cmd/assessMigrationCommand.go

```diff
@@ -63,7 +65,21 @@ var assessMigrationCmd = &cobra.Command{
 		validateSourceSchema()
 		validatePortRange()
 		validateSSLMode()
-		validateAssessmentMetadataDirFlag()
+		if cmd.Flags().Changed("assessment-metadata-dir") {
+			validateAssessmentMetadataDirFlag()
+			cmd.Flags().VisitAll(func(f *pflag.Flag) {
+				if f.Changed {
```

Comment:
While this might work, I'd recommend better list out all the flags that we want to check explicitly :) It'd be less confusing and more clear.

Reply:
I thought this might be better than listing all the `source-*` flags as those are in the majority and we need to just exclude `--source-db-type` only.

Reply:
(btw, this is a nit, don't block PR on this :D just discussing because of interest)
As a general thumb rule, [explicit tends to be better than implicit.](https://peps.python.org/pep-0020/) The less complicated the logic, the better.  The code becomes more readable as a result. I think one of golang's goals also is simplicity. 

In this example, if you list all the flags here, you immediately find out what all flags are being disallowed. The code becomes very clear as opposed to understanding the condition (containing "source") and then going to the place in the code where the flags are declared. 

Ofc you can argue from the other side - If we add more flags in the future, this logic will automatically incorporate it (but also maybe it might incorrectly block a flag from being passed, and we'll have to come change it.. )

There's no right answer..

Reply:
hmm okay, makes sense in terms of readability. Will make the change. Thanks!